suite_context.go

```diff
@@ -121,6 +164,12 @@ func (s *suiteContext) ResetBeforeEachScenario(interface{}) {
 	SuiteContext(s.testedSuite, s.extraCIs...)
 	// reset all fired events
 	s.events = []*firedEvent{}
+	s.allowInjection = oldAllowInjection
```

Comment:
should `allowInjection` be reset to `false`?

Reply:
it can be reset. I keep the initial comportment of the variable but if you rather have it reset, it can be

Reply:
Also though all of s fields where reseted ^^

Reply:
yea, I don't think we should keep state between scenarios if there isn't a clear purpose for doing so

Reply:
Done